lib/vmdb/plugins.rb

```diff
@@ -4,29 +4,47 @@ class Plugins
 
     attr_reader :vmdb_plugins
     attr_reader :registered_automate_domains
+    attr_reader :registered_provider_plugins
 
     def initialize
       @registered_automate_domains = []
+      @registered_provider_plugins = []
       @vmdb_plugins = []
     end
 
     def register_vmdb_plugin(engine)
       @vmdb_plugins << engine
 
       register_automate_domains(engine)
+      register_provider_plugin(engine)
 
       # make sure STI models are recognized
       DescendantLoader.instance.descendants_paths << engine.root.join('app')
     end
 
-    def register_automate_domains(engine)
-      Dir.glob(engine.root.join("content", "automate", "*")).each do |domain_directory|
-        @registered_automate_domains << AutomateDomain.new(domain_directory)
+    def provider_plugin_names
+      @provider_plugins_names ||= registered_provider_plugins.collect do |plugin|
+        ManageIQ::Providers::Inflector.provider_name(plugin).downcase.to_sym
```

Comment:
Should this be `underscore` rather than `downcase` to deal with things like `AnsibleTower` in the future?